libs/utils/decorate.js

```diff
@@ -325,20 +393,77 @@ export async function loadCDT(el, classList) {
   }
 }
 
+export function isAccessible(anchorTag) {
+  return !anchorTag.hash.includes(HIDE_CONTROLS);
+}
+
+function updateFirstVideo() {
+  if (firstVideo != null && firstVideo?.controls === false && videoCounter > 1) {
+    let videoHolder = document.querySelector('[video-index="1"]') || firstVideo.closest('.video-holder');
+    if (videoHolder.nodeName !== 'A') {
+      videoHolder = videoHolder.querySelector('a.pause-play-wrapper');
+    }
+    const firstVideoLabel = videoHolder.getAttribute('aria-label');
+    videoHolder.setAttribute('aria-label', `${firstVideoLabel} 1`);
+    firstVideo = null;
+  }
+}
+
+function updateAriaLabel(videoEl, videoAttrs) {
+  if (!videoEl.getAttributeNames().join().includes('hoverplay')) {
+    const pausePlayWrapper = videoEl.parentElement.querySelector('.pause-play-wrapper') || videoEl.closest('.pause-play-wrapper');
+    const indexOfVideo = pausePlayWrapper.getAttribute('video-index');
+    const ariaLabel = `${videoAttrs.includes('autoplay') ? labels.pauseMotion : labels.playMotion} ${indexOfVideo === '1' && videoCounter === 1 ? '' : indexOfVideo}`.trim();
+    pausePlayWrapper.setAttribute('aria-label', ariaLabel);
+    updateFirstVideo();
+  }
+}
+
+export function decoratePausePlayWrapper(videoEl, videoAttrs) {
+  if (labels === null) {
+    replaceKeyArray(['pause-motion', 'play-motion', 'pause-icon', 'play-icon'], getFedsPlaceholderConfig())
+      .then(([pauseMotion, playMotion, pauseIcon, playIcon]) => {
+        labels = { playMotion, pauseMotion, pauseIcon, playIcon };
+        updateAriaLabel(videoEl, videoAttrs);
+      });
+  } else {
+    updateAriaLabel(videoEl, videoAttrs);
+  }
+}
+
 export function decorateAnchorVideo({ src = '', anchorTag }) {
   if (!src.length || !(anchorTag instanceof HTMLElement)) return;
+  const accessibilityEnabled = isAccessible(anchorTag);
+  anchorTag.hash = anchorTag.hash.replace(`#${HIDE_CONTROLS}`, '');
   if (anchorTag.closest('.marquee, .aside, .hero-marquee, .quiz-marquee') && !anchorTag.hash) anchorTag.hash = '#autoplay';
   const { dataset, parentElement } = anchorTag;
-  const video = `<video ${getVideoAttrs(anchorTag.hash, dataset)} data-video-source=${src}></video>`;
+  const attrs = getVideoAttrs(anchorTag.hash, dataset);
+  const tabIndex = anchorTag.tabIndex || 0;
+  const videoIndex = (tabIndex === -1) ? 'tabindex=-1' : '';
+  let video = `<video ${attrs} data-video-source=${src} ${videoIndex}></video>`;
+  videoCounter += 1;
```

Comment:
Should we increase the counter only for videos that actually get the controls?

Reply:
would really matter? since we just need a unique identifier for the video, should we add an extra condition?